models/optional_modules/consent/scratch/default/snowplow_web_consent_events_this_run.sql

```diff
@@ -23,7 +23,8 @@ with prep as (
     p.consent_scopes,
     p.domains_applied,
     p.gdpr_applies,
-    v.elapsed_time as cmp_load_time
+    v.elapsed_time as cmp_load_time,
+    row_number() over (partition by e.event_id order by e.load_tstamp) as dedupe_index
```

Comment:
I think this technically will work (as you're still doing the row_number after the join), but it might be more efficient given each context table is sortkey on date to split them out into subquieries first and de-dupe on them first, then join on after.